src/ImageSharp/Advanced/AdvancedImageExtensions.cs

```diff
@@ -27,7 +27,7 @@ public static IImageEncoder DetectEncoder(this Image source, string filePath)
         Guard.NotNull(filePath, nameof(filePath));
 
         string ext = Path.GetExtension(filePath);
-        IImageFormat format = source.GetConfiguration().ImageFormatsManager.FindFormatByFileExtension(ext);
+        IImageFormat? format = source.GetConfiguration().ImageFormatsManager.FindFormatByFileExtension(ext);
```

Comment:
Ah... This is one of the areas I want to have a look at. Ideally this method should be rewritten to use the try-out pattern.

e.g 

``` c#
bool ImageFormatsManager.TryFindFormatByFileExtension(ext, [NotNullWhen(true)] out IImageFormat? format)
```

We'd return named tuples for async equivalents.

Reply:
Ok gonna rewrite that.

Reply:
Legend. Thanks! It's something I very much wish we'd done from the start.